File: pkg/types/record.go
package types

type RecordRef struct {
	ID         string
	DatabaseID string
	FormID     string
}

type FormRef struct {
	DatabaseID string
	FormID     string
}

type Record struct {
	ID         string                 `json:"id"`
	Seq        int64                  `json:"seq"`
	DatabaseID string                 `json:"databaseId"`
	FormID     string                 `json:"formId"`
	ParentID   *string                `json:"parentId"`
	Values     map[string]interface{} `json:"values"`
}

type RecordList struct {
	Items []*Record `json:"items"`
}

type RecordListOptions struct {
	DatabaseID string
	FormID     string
}

File: frontend/packages/core-api-client/src/lib/types/model/Record.ts
export type FieldValue = {
  fieldId: string;
  value: string | string[] | null;
};

export type Record = {
  id: string;
  databaseId: string;
  formId: string;
  ownerId: string | undefined;
  values: FieldValue[];
};

export type LocalRecord = Record & {
  isNew: boolean;
};

export type RecordList = { items: Record[] };
